plugins/content-manager/src/main/java/com/wazuh/contentmanager/updater/ContentUpdater.java

```diff
@@ -0,0 +1,171 @@
+     */
+    public void fetchAndApplyUpdates(Long fixedOffset) throws ContentUpdateException {
+        Long currentOffset = this.getCurrentOffset();
+        Long lastOffset = this.getLatestOffset();
+
+        // Placeholder for testing purposes. TODO: Remove.
+        if (fixedOffset != null) {
+            currentOffset = fixedOffset;
+        }
+
+        if (lastOffset <= currentOffset) {
+            log.info("No new updates available. Current offset ({}) is up to date.", currentOffset);
+            return;
+        }
+
+        log.info("New offsets available updating to offset: {}", lastOffset);
+        while (currentOffset < lastOffset) {
+            Long nextOffset = Math.min(currentOffset + CHUNK_MAX_SIZE, lastOffset);
+            ContextChanges changes =
+                    this.getContextChanges(currentOffset.toString(), nextOffset.toString());
+            log.info("Fetched offsets from {} to {}", currentOffset, nextOffset);
+
+            // If there is an error fetching the changes, stop the process.
+            if (changes == null) {
+                throw new ContentUpdateException(
+                        "Error fetching changes for offsets " + currentOffset + " to " + nextOffset, null);
+            }
+            // Apply the fetched changes to the indexed context.
+            if (!this.patchContextIndex(changes)) {
+                this.restartConsumerInfo();
+                return;
+            }
+
+            currentOffset = nextOffset;
+        }
+
+        this.postUpdateCommand();
+    }
+
+    /**
+     * Fetches the context changes between a given offset range from the CTI API.
+     *
+     * @param fromOffset Starting offset (inclusive).
+     * @param toOffset Ending offset (exclusive).
+     * @return ContextChanges object containing the changes.
+     */
+    @VisibleForTesting
+    ContextChanges getContextChanges(String fromOffset, String toOffset) {
+        return Privileged.doPrivilegedRequest(
+                () -> CTIClient.getInstance().getChanges(fromOffset, toOffset, null));
+    }
+
+    /**
+     * Retrieves the latest offset from the CTI API.
+     *
+     * @return Latest available offset.
+     */
+    @VisibleForTesting
+    Long getLatestOffset() {
+        ConsumerInfo consumerInfo =
+                Privileged.doPrivilegedRequest(() -> CTIClient.getInstance().getCatalog());
+        return consumerInfo.getLastOffset();
+    }
+
+    /**
+     * Retrieves the current stored "last" offset from {@link ContextIndex}.
+     *
+     * @return The current "last" offset.
+     */
+    @VisibleForTesting
+    Long getCurrentOffset() {
+        ConsumerInfo consumerInfo =
+                Privileged.doPrivilegedRequest(
+                        () ->
+                                ContextIndex.getInstance()
+                                        .getConsumer(PluginSettings.CONTEXT_ID, PluginSettings.CONSUMER_ID));
```

Comment:
Is it required to use "privileged" to search on a non-protected index?

Reply:
This is similar to https://github.com/wazuh/wazuh-indexer-plugins/pull/307#discussion_r2026655205, when not using privileges it failed with 
```
[2025-04-01T14:29:06,979][ERROR][c.w.c.i.ContextIndex     ] [integTest-0] Failed to retrieve CTI Catalog Context vd_4.8.0, Exception: java.security.AccessControlException: access denied ("java.util.PropertyPermission" "opensearch.experimental.feature.tiered_remote_index.enabled" "read")
```